geometric_kernels/frontends/pytorch/gpytorch.py

```diff
@@ -1,63 +1,96 @@
 """
-Pytorch kernel wrapper
+GPyTorch kernel wrapper
 """
 import gpytorch
-import numpy as np
 import torch
 
 from geometric_kernels.kernels import BaseGeometricKernel
 from geometric_kernels.spaces.base import Space
 
 
-class GPytorchGeometricKernel(gpytorch.kernels.Kernel):
+class GPyTorchGeometricKernel(gpytorch.kernels.Kernel):
     """
-    Pytorch wrapper for `BaseGeometricKernel`
+    GPyTorch wrapper for `BaseGeometricKernel`
+
+    Note: you cannot change trainable_nu after constructing the object.
     """
 
     has_lengthscale = True
 
     def __init__(
         self,
-        kernel: BaseGeometricKernel,
-        lengthscale=1.0,
-        nu=np.inf,
+        base_kernel: BaseGeometricKernel,
+        lengthscale=None,
+        nu=None,
+        variance=1.0,
         trainable_nu: bool = False,
         **kwargs,
     ):
         super().__init__(**kwargs)
 
-        self._kernel = kernel
+        self.base_kernel = base_kernel
+
+        default_params = base_kernel.init_params()
+
+        if nu is None:
+            nu = default_params["nu"]
 
-        self.lengthscale = torch.tensor(lengthscale)
+        if lengthscale is None:
+            lengthscale = default_params["lengthscale"]
 
-        self.trainable_nu = trainable_nu
-        if self.trainable_nu and torch.isinf(nu):
+        lengthscale = torch.as_tensor(lengthscale)
+        variance = torch.as_tensor(variance)
+        nu = torch.as_tensor(nu)
+
+        self._trainable_nu = trainable_nu
+        if self._trainable_nu and torch.isinf(nu):
             raise ValueError("Cannot have trainable `nu` parameter with infinite value")
 
-        if self.tranable_nu:
+        self.lengthscale = lengthscale
+
+        self.register_parameter(
+            name="raw_variance", parameter=torch.nn.Parameter(torch.tensor(1.0))
+        )
+        self.register_constraint("raw_variance", gpytorch.constraints.Positive())
+        self.variance = variance
+
+        if self._trainable_nu:
             self.register_parameter(
-                name="raw_nu", parameter=torch.nn.Parameter(torch.tensor(nu))
+                name="raw_nu", parameter=torch.nn.Parameter(torch.tensor(1.0))
```

Comment:
Why ignore user-provided `nu`?

Reply:
The user-provided `nu` is used right below. The thing is here we initialize `raw_new`, which is a transformed version of the vanilla `nu`. Not to get through this transofrmation manually, we first set it with a dummy value and then use a smart setter to accommodate the user-provided `nu`.